samza-core/src/main/scala/org/apache/samza/container/SamzaContainerMetrics.scala

```diff
@@ -49,6 +49,7 @@ class SamzaContainerMetrics(
   val physicalMemoryMb = newGauge("physical-memory-mb", 0.0F)
   val physicalMemoryUtilization = newGauge("physical-memory-utilization", 0.0F)
   val containerThreadPoolSize = newGauge("container-thread-pool-size", 0L)
```

Comment:
Given you are updating as part of `hostStatisticsMonitor`,  you will need to remove the code in `SamzaContainer` which updates this metric on startup to `job.container.thread.pool.size`.  This would prevent any weird metric drop given the `poolSize` only gets to this upper bound eventually and when there are not enough new tasks submitted, it may not end up creating additional threads.

Reply:
@mynameborat I already removed `samzaContainerMetrics.containerThreadPoolSize.set(threadPoolSize)` in SamzaContainer.scala. `hostStatisticsMonitor` is the only place we set the `containerThreadPoolSize` metric value

Reply:
Cool. I must have looked at the stale commit.